homeassistant/components/rejseplanen/coordinator.py

```diff
@@ -0,0 +1,178 @@
+"""Data update coordinator for Rejseplanen."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+
+from py_rejseplan.api.departures import DepartureBoard, departuresAPIClient
+from py_rejseplan.dataclasses.departure import Departure
+from py_rejseplan.exceptions import api_error, connection_error, http_error
+from py_rejseplan.version import __version__ as py_rejseplan_version
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN, SCAN_INTERVAL_MINUTES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RejseplanenDataUpdateCoordinator(DataUpdateCoordinator[DepartureBoard | None]):
+    """Class to manage fetching data from the Rejseplanen API."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
```

Comment:
We should use the extended config entry type here. You might need to move the type declaration to this file